mypyc/ir/ops.py

```diff
@@ -1252,6 +1252,70 @@ def accept(self, visitor: 'OpVisitor[T]') -> T:
         return visitor.visit_load_global(self)
 
 
+class BinaryIntOp(RegisterOp):
+    """Binary operations on integer types
+
+    These ops are low-level and will be eventually generated to simple x op y form.
+    The left and right values should be of low-level integer types that support those ops
+    """
+    error_kind = ERR_NEVER
+
+    # arithmetic
+    ADD = 0  # type: Final
+    SUB = 1  # type: Final
+    MUL = 2  # type: Final
+    DIV = 3  # type: Final
+    MOD = 4  # type: Final
+    # logical
+    EQ = 100  # type: Final
+    NEQ = 101  # type: Final
+    LT = 102  # type: Final
+    GT = 103  # type: Final
+    LEQ = 104  # type: Final
+    GEQ = 105  # type: Final
+    # bitwise
+    AND = 200  # type: Final
+    OR = 201  # type: Final
+    XOR = 202  # type: Final
+    LEFT_SHIFT = 203  # type: Final
+    RIGHT_SHIFT = 204  # type: Final
+
+    op_str = {
+        ADD: '+',
+        SUB: '-',
+        MUL: '*',
+        DIV: '/',
+        MOD: '%',
+        EQ: '==',
+        NEQ: '!=',
+        LT: '<',
+        GT: '>',
+        LEQ: '<=',
+        GEQ: '>=',
+        AND: '&',
+        OR: '|',
+        XOR: '^',
+        LEFT_SHIFT: '<<',
+        RIGHT_SHIFT: '>>',
+    }  # type: Final
+
+    def __init__(self, type: RType, lhs: Value, rhs: Value, op: int, line: int = -1) -> None:
```

Comment:
It would be good to store the integer type also here, since different code may be generated for 32 bit and 64 bit adds, for example, on non-C back ends.

Reply:
Actually, it's better to do this in a separate PR, and also update things to more generally store the result type.